04-a-catalogue-of-pedagogical-patterns.md

```diff
@@ -255,6 +255,51 @@ Some students don't find this approach engaging. In particular, if the
 exercise is too simple for the level of programming competency of the
 students, it can be perceived a "make-work" task.
 
+## Target Practice
+
+
+**Description:**<br/>
+This pattern is focusses attention on one component of a multi-step workflow. 
+Learners are asked to implement one stage of a workflow while preceeding and 
+subsequent stages may be provided to them. 
+
+
+**Example:**<br/>
+In a climate-science exercise, the notebook that is provided to students 
+provides code for fetching and parsing 20 years of hourly average temperature 
+data from a public database. Students are asked to design an algorithm for computing
+yearly average temperature and standard deviation. Plotting code which plots the
+yearly temperature with error-bars showing the standard deviation. 
+
+**Related patterns:**<br/>
+This pattern is similar to [Fill in the Blank] exercises. Fill in the Blank
+exercises are typically smaller and more immediate, while Target Practice
+exercises tend to be larger (e.g. an entire step in a multi-step process). 
+
+
+**Learning goals:**<br/>
+The aim of a Target Practice exercise is to focus attention on one 
+component of a workflow and practice skills for solving that component. 
+
```

Comment:
It can also be used to reflect on broader consequences of choices made in the target step of an integrated workflow/analysis.